app/src/main/java/chat/rocket/android/helper/MessageParser.kt

```diff
@@ -0,0 +1,176 @@
+                                    private val builder: SpannableBuilder,
+                                    private val senderNameLength: Int) : SpannableMarkdownVisitor(configuration, builder) {
+
+        override fun visit(blockQuote: BlockQuote) {
+
+            // mark current length
+            val length = builder.length()
+
+            // pass to super to apply markdown
+            super.visit(blockQuote)
+
+            val res = context.resources
+            val timeOffsetStart = length + senderNameLength + 1
+            builder.setSpan(QuoteMarginSpan(context.getDrawable(R.drawable.quote), 10), length, builder.length())
+            builder.setSpan(StyleSpan(Typeface.BOLD), length, length + senderNameLength)
+            builder.setSpan(ForegroundColorSpan(Color.BLACK), length, builder.length())
+            // set time spans
+            builder.setSpan(AbsoluteSizeSpan(res.getDimensionPixelSize(R.dimen.message_time_text_size)),
+                    timeOffsetStart, builder.length())
+            builder.setSpan(ForegroundColorSpan(res.getColor(R.color.darkGray)),
+                    timeOffsetStart, builder.length())
+        }
+    }
+
+    class QuoteMessageBodyVisitor(private val context: Context,
+                                  configuration: SpannableConfiguration,
+                                  private val builder: SpannableBuilder) : SpannableMarkdownVisitor(configuration, builder) {
+
+        override fun visit(blockQuote: BlockQuote) {
+
+            // mark current length
+            val length = builder.length()
+
+            // pass to super to apply markdown
+            super.visit(blockQuote)
+
+            builder.setSpan(QuoteMarginSpan(context.getDrawable(R.drawable.quote), 10), length, builder.length())
+        }
+    }
+
+    class QuoteMarginSpan(quoteDrawable: Drawable, private var pad: Int) : LeadingMarginSpan, LineHeightSpan {
+        private var drawable: Drawable = quoteDrawable
+
+        override fun getLeadingMargin(first: Boolean): Int {
+            return drawable.intrinsicWidth + pad
+        }
+
+        override fun drawLeadingMargin(c: Canvas, p: Paint, x: Int, dir: Int,
+                                       top: Int, baseline: Int, bottom: Int,
+                                       text: CharSequence, start: Int, end: Int,
+                                       first: Boolean, layout: Layout) {
+            val st = (text as Spanned).getSpanStart(this)
+            val ix = x
+            val itop = layout.getLineTop(layout.getLineForOffset(st))
+            val dw = drawable.intrinsicWidth
+            val dh = drawable.intrinsicHeight
+            // XXX What to do about Paint?
+            drawable.setBounds(ix, itop, ix + dw, itop + layout.height)
+            drawable.draw(c)
+        }
+
+        override fun chooseHeight(text: CharSequence, start: Int, end: Int,
+                                  istartv: Int, v: Int,
```

Comment:
what `istartv` means? And I think it has camelCase problems

Reply:
It's spanstartv

Reply:
Those parameters are for measureing. It's `v`for vertical and `spanstartv` on the `LineHeightSpan` interface. I don't think it would look meaningful in camelcase.